cstore_fdw.c

```diff
@@ -300,6 +301,9 @@ CStoreProcessUtility(Node * parseTree, const char *queryString,
 		List *allTablesList = truncateStatement->relations;
 		List *cstoreTablesList = FindCStoreTables(allTablesList);
 		List *otherTablesList = list_difference(allTablesList, cstoreTablesList);
+
+		CheckTruncatePermissions(cstoreTablesList);
```

Comment:
If permissions of a table change by a concurrent session between call to CheckTruncatePermissions() and TruncateCStoreTables(), do we see it here?

(1) PostgreSQL's ExecuteTruncate() seems to do permission check & truncation while holding an AccessExclusiveLock, so I guess permissions cannot change after permission check and before end of TRUNCATE.
(2) Not caused by this change, but PostgreSQL grabs a lock on *all* relations before starting truncating the first table. In cstore_fdw we lock, truncate, release tables one by one.

So our behaviour here is not similar to PostgreSQL here.  I don't think this is a big issue, just trying to understand.

@anarazel what do you think? Can not having same semantics as PostgreSQL here cause much harm? (I'm sure these 2 aren't the only issues)